### PR TITLE
Add a PhoneNumber class to handle formatting and actions on numbers

### PR DESCRIPTION
**Pull request by tdupreeibp**
We value collaboration at Invoca, and focus on peer mentorship as a way for the whole team to learn and thrive. As part of this process, we conduct code reviews during our development cycle. In this phase of the interview, we would like you to review and comment on this Pull Request. 

You can assume that this Pull Request was written by an intern, who is new to professional software development.  Please take about 20-30 minutes to review

### CODE REVIEW

File: phone_number.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  def text_and_call(message)
+    PhoneActions.new(USERNAME, PASSWORD).text!(@number, message)
+    PhoneActions.new(USERNAME, PASSWORD).call!(message, message)
+  end
+
+  def number
+    @number
+  end
```
Comment (ap2322) [inline phone_number.rb:27]: Opportunity for `attr_reader` syntactic sugar here

File: phone_number_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+require_relative "./phone_number"
+
+RSpec.describe PhoneNumber do
+  let(:phone_number) { PhoneNumber.new("5558675309") }
+
+  describe "#number" do
+    it "is a formatted number" do
+      expect(phone_number.number).to eq("(555) 867-5309")
+    end
+  end
```
Comment (ap2322) [inline phone_number_spec.rb:10]: This would be a good opportunity to test a variety of different strings given as the `number` to look for edge cases if the string isn't formatted exactly as expected.
- with parenthesis
- one parenthesis
- unexpected alpha characters

File: phone_number.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  def to_i
+    n = @number.gsub!(/[^0-9]/, "")
+    n.to_i
+  end
```
Comment (ap2322) [inline phone_number.rb:32]: Try not to use the same method name as ruby string method `#to_i`. Looks a bit confusing also because `to_i` gets called within the method as well.

File: phone_number_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  describe "#text_and_call" do
+    it "calls both PhoneActions#text! and PhoneActions#call! methods" do
+      expect_any_instance_of(PhoneActions).to receive(:text!)
+      expect_any_instance_of(PhoneActions).to receive(:call!)
+      phone_number.text_and_call("hello world!")
+    end
+  end
```
Comment (ap2322) [inline phone_number_spec.rb:39]: Instead of stubbing every instance of PhoneActions try to split out the PhoneActions spec file and call it more directly to get more granular testing. It doesn't look like the `call` or `text` methods of either class are getting fully tested.

File: phone_number.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  def call(message)
+    PhoneActions.new(USERNAME, PASSWORD).call!(@number, message)
+  end
+
+  def text_and_call(message)
+    PhoneActions.new(USERNAME, PASSWORD).text!(@number, message)
+    PhoneActions.new(USERNAME, PASSWORD).call!(message, message)
+  end
```
Comment (ap2322) [inline phone_number.rb:23]: Because USERNAME and PASSWORD are only used to pass through to the PhoneActions class, there's an opportunity to think through what belongs in PhoneNumber and PhoneActions and create a more clear boundary between the two classes.

File: phone_number.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
+require_relative "./phone_actions"
+
+class PhoneNumber
+  USERNAME = ENV['ACTIONS_USERNAME']
+  PASSWORD = ENV['ACTIONS_PASSWORD']
+
+  def initialize(number)
+    @number = number
+    @number = format_number
```
Comment (ap2322) [inline phone_number.rb:9]: Opportunity to clean up the duplicate setting of `@number` here.

File: phone_actions.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+class PhoneActions
```
Comment (ap2322) [inline phone_actions.rb:1]: Is this class intended as a service class or controller to make REST calls? These actions look a bit unconventional.

**Review by ap2322 — CHANGES_REQUESTED** [review]
This is good start to adding some key functionality. My comments highlight some ways you can more clearly organize your classes and spec files. Also, I've pointed out some conventions, clarifications, and spec additions you can use to make your code more readable and reliable.

I would be happy to take a look at this ticket with you and spend some time pairing to refactor and fill in spec files. This is a good start, let's polish it up.
